svf/lib/SVFIR/SymbolTableInfo.cpp

```diff
@@ -105,10 +105,23 @@ LocationSet SymbolTableInfo::getModulusOffset(const MemObj* obj, const LocationS
         offset = abs(offset);
     }
     u32_t maxOffset = obj->getMaxFieldOffsetLimit();
-    if (maxOffset != 0)
-        offset = offset % maxOffset;
+
+    if (Options::CyclicFldIdx())
+    {
```

Comment:
Can you put two examples to explain these two branch choices

Reply:
Here is the result on perlbmk

wpa -ander perlbmk.orig -cyclic-field-index=true

*********CallGraph Stats***************
################ (program : perlbmk.orig)###############
-------------------------------------------------------
TotalNode           1168
TotalEdge           9816
TotalCycle          4
MaxNodeInCycle      832
NodeInCycle         835
CalRetPairInCycle   7254
#######################################################

*********General Stats***************
################ (program : perlbmk.orig)###############
BBWith2Succ         11697
BBWith3Succ         146
TotalPointers       206573
TotalObjects        8215
TotalFieldObjects   295
MaxStructSize       29
TotalSVFStmts       201323
TotalPTASVFStmts    112382
FIObjNum            584
FSObjNum            7630
AddrsNum            11436
LoadsNum            59431
StoresNum           23970
CopysNum            14111
GepsNum             26553
CallsNum            12145
ReturnsNum          4610
FunctionObjs        1168
GlobalObjs          415
HeapObjs            129
StackObjs           6501
VarStructObj        70
VarArrayObj         91
ConstStructObj      24
ConstArrayObj       10
NonPtrObj           3402
ConstantObj         0
IndCallSites        57
TotalCallSite       8441
-------------------------------------------------------
LLVMIRTime          0.826
SymbolTableTime     0.305
SVFIRTime           26.69
#######################################################

*********Constraint Graph Stats***************
################ (program : perlbmk.orig)###############
-------------------------------------------------------
AvgIn/OutCopyEdge   0.914517
AvgIn/OutLoadEdge   0.455909
AvgIn/OutAddrEdge   0.0981527
AvgIn/OutEdge       1.58589
LocalVarInRecur     2434
NumOfCGNode         197419
TotalValidNode      78928
TotalValidObjNode   5740
NumOfCGEdge         117424
NumOfAddrs          7747
NumOfCopys          47797
NumOfGeps           24384
NumOfLoads          35984
NumOfStores         9259
MaxInCopyEdge       725
MaxOutCopyEdge      20750
MaxInLoadEdge       4747
MaxOutLoadEdge      847
MaxInStoreEdge      348
MaxOutStoreEdge     3421
AvgIn/OutStoreEdge  0
MaxInAddrEdge       1
MaxOutAddrEdge      2069
#######################################################

*********Andersen Pointer Analysis Stats***************
################ (program : perlbmk.orig)###############
-------------------------------------------------------
TotalTime           34.67
SCCDetectTime       0
SCCMergeTime        0
LoadStoreTime       0
CopyGepTime         0
UpdateCGTime        0
AvgPtsSetSize       176.801
AvgTopLvlPtsSize    401.47
CollapseTime        0
TotalPointers       206573
Pointers            206449
TotalObjects        8510
IndCallSites        57
AddrProcessed       10282
CopyProcessed       272988
GepProcessed        72011
LoadProcessed       2637566
StoreProcessed      348794
NumOfSFRs           0
NumOfFieldExpand    0
MemObjects          8215
DummyFieldPtrs      124
FieldObjs           295
MaxPtsSetSize       569
SolveIterations     14
IndEdgeSolved       1410
NumOfSCCDetect      14
TotalCycleNum       234
TotalPWCCycleNum    24
NodesInCycles       4981
MaxNodesInSCC       2771
NullPointer         162
PointsToConstPtr    20790
PointsToBlkPtr      0
#######################################################

****Persistent Points-To Cache Statistics: Andersen's analysis bitvector****
################ (program : perlbmk.orig)###############
UniquePointsToSets       10754
TotalUnions              398426
PropertyUnions           223298
UniqueUnions             2064
LookupUnions             169403
PreemptiveUnions         3661
TotalComplements         2840313
PropertyComplements      2676921
UniqueComplements        904
LookupComplements        161585
PreemptiveComplements    903
TotalIntersections       394766
PropertyIntersections    392956
UniqueIntersections      1
LookupIntersections      3
PreemptiveIntersections  1806
#######################################################


wpa -ander perlbmk.orig -cyclic-field-index=false

*********CallGraph Stats***************
################ (program : perlbmk.orig)###############
-------------------------------------------------------
TotalNode           1168
TotalEdge           9816
TotalCycle          4
MaxNodeInCycle      832
NodeInCycle         835
CalRetPairInCycle   7254
#######################################################

*********General Stats***************
################ (program : perlbmk.orig)###############
BBWith2Succ         11697
BBWith3Succ         146
TotalPointers       206573
TotalObjects        8215
TotalFieldObjects   295
MaxStructSize       29
TotalSVFStmts       201323
TotalPTASVFStmts    112382
FIObjNum            584
FSObjNum            7630
AddrsNum            11436
LoadsNum            59431
StoresNum           23970
CopysNum            14111
GepsNum             26553
CallsNum            12145
ReturnsNum          4610
FunctionObjs        1168
GlobalObjs          415
HeapObjs            129
StackObjs           6501
VarStructObj        70
VarArrayObj         91
ConstStructObj      24
ConstArrayObj       10
NonPtrObj           3402
ConstantObj         0
IndCallSites        57
TotalCallSite       8441
-------------------------------------------------------
LLVMIRTime          0.824
SymbolTableTime     0.313
SVFIRTime           27.479
#######################################################

*********Constraint Graph Stats***************
################ (program : perlbmk.orig)###############
-------------------------------------------------------
AvgIn/OutCopyEdge   0.914517
AvgIn/OutLoadEdge   0.455909
AvgIn/OutAddrEdge   0.0981527
AvgIn/OutEdge       1.58589
LocalVarInRecur     2434
NumOfCGNode         197419
TotalValidNode      78928
TotalValidObjNode   5740
NumOfCGEdge         117424
NumOfAddrs          7747
NumOfCopys          47797
NumOfGeps           24384
NumOfLoads          35984
NumOfStores         9259
MaxInCopyEdge       725
MaxOutCopyEdge      20750
MaxInLoadEdge       4747
MaxOutLoadEdge      847
MaxInStoreEdge      348
MaxOutStoreEdge     3421
AvgIn/OutStoreEdge  0
MaxInAddrEdge       1
MaxOutAddrEdge      2069
#######################################################

*********Andersen Pointer Analysis Stats***************
################ (program : perlbmk.orig)###############
-------------------------------------------------------
TotalTime           30.41
SCCDetectTime       0
SCCMergeTime        0
LoadStoreTime       0
CopyGepTime         0
UpdateCGTime        0
AvgPtsSetSize       176.801
AvgTopLvlPtsSize    401.47
CollapseTime        0
TotalPointers       206573
Pointers            206449
TotalObjects        8510
IndCallSites        57
AddrProcessed       10282
CopyProcessed       271494
GepProcessed        72011
LoadProcessed       2637226
StoreProcessed      348758
NumOfSFRs           0
NumOfFieldExpand    0
MemObjects          8215
DummyFieldPtrs      124
FieldObjs           295
MaxPtsSetSize       569
SolveIterations     14
IndEdgeSolved       1410
NumOfSCCDetect      14
TotalCycleNum       234
TotalPWCCycleNum    24
NodesInCycles       4981
MaxNodesInSCC       2771
NullPointer         162
PointsToConstPtr    20790
PointsToBlkPtr      0
#######################################################

****Persistent Points-To Cache Statistics: Andersen's analysis bitvector****
################ (program : perlbmk.orig)###############
UniquePointsToSets       10583
TotalUnions              396769
PropertyUnions           223209
UniqueUnions             1986
LookupUnions             168072
PreemptiveUnions         3502
TotalComplements         2838885
PropertyComplements      2676016
UniqueComplements        796
LookupComplements        161278
PreemptiveComplements    795
TotalIntersections       374025
PropertyIntersections    372431
UniqueIntersections      1
LookupIntersections      3
PreemptiveIntersections  1590
#######################################################